cache.h

```diff
@@ -1887,4 +1887,10 @@ int print_sha1_ellipsis(void);
 /* Return 1 if the file is empty or does not exists, 0 otherwise. */
 int is_empty_or_missing_file(const char *filename);
 
+/*
+ * Extended crc32 with 64 bit address range
+ * On Windows, uInt/uLong are only 32 bits.
+ */
+extern uLong xcrc32(uLong crc, const unsigned char *buf, size_t bytes);
+
```

Comment:
The `crc32()` function is declared and defined in zlib. It would be most helpful to mention this in the code comment, along with the hint that `zlib.h` is already included in `cache.h`.

Reply:
Hmm, I had though that we got it from the gnu stuff (i.e. I found it's definition from the gnu ref pages, not the zlib pages).
Double check needed. It's good if you are right (probably are..)

Reply:
Looks like I was misinformed. The crc32 is part of the zlib library and I had missed that fact.

Reply:
TBH I did not know either until I looked which C standard defines the `crc32()` function, and did not find any.